x-pack/legacy/plugins/siem/server/lib/detection_engine/alerts/utils.test.ts

```diff
@@ -0,0 +1,658 @@
+      // need a sample search result, sample signal params, mock service, mock logger
+      const sampleParams = sampleSignalAlertParams(undefined);
+      const sampleSearchResult = sampleDocSearchResultsNoSortId;
+      const savedObjectsClient = savedObjectsClientMock.create();
+      const bulkBody = sampleDocSearchResultsNoSortId.hits.hits.flatMap(doc => [
+        {
+          index: {
+            _index: process.env.SIGNALS_INDEX || '.siem-signals-10-01-2019',
+            _id: doc._id,
+          },
+        },
+        buildBulkBody(doc, sampleParams),
+      ]);
+      const mockService: AlertServices = {
+        callCluster: async (action: string, params: BulkIndexDocumentsParams) => {
+          expect(action).toEqual('bulk');
+          expect(params.index).toEqual('.siem-signals-10-01-2019');
+
+          // timestamps are annoying...
+          (bulkBody[1] as SignalHit).signal['@timestamp'] = params.body[1].signal['@timestamp'];
+          expect(params.body).toEqual(bulkBody);
+          return {
+            took: 100,
+            errors: true,
+          };
+        },
+        alertInstanceFactory: jest.fn(),
+        savedObjectsClient,
+      };
+      const successfulSingleBulkIndex = await singleBulkIndex(
+        sampleSearchResult,
+        sampleParams,
+        mockService,
+        mockLogger
+      );
+      expect(mockLogger.debug).toHaveBeenCalledTimes(2);
+      expect(mockLogger.warn).toHaveBeenCalledTimes(0);
+      expect(mockLogger.error).toHaveBeenCalledTimes(1);
+      expect(successfulSingleBulkIndex).toEqual(false);
+    });
+  });
+  describe('singleSearchAfter', () => {
+    test('if singleSearchAfter works without a given sort id', async () => {
+      let searchAfterSortId;
+      const sampleParams = sampleSignalAlertParams(undefined);
+      const savedObjectsClient = savedObjectsClientMock.create();
+      const expectedSearchAfterQuery = buildEventsSearchQuery({
+        index: sampleParams.index,
+        from: sampleParams.from,
+        to: sampleParams.to,
+        filter: sampleParams.filter,
+        size: sampleParams.size ? sampleParams.size : 1,
+        maxDocs: undefined,
+        searchAfterSortId: undefined,
+      });
+      const mockService: AlertServices = {
+        callCluster: async (action: string, params: SearchParams) => {
+          expect(action).toEqual('search');
+          expect(params.index).toEqual(sampleParams.index);
+          expect(params).toEqual(expectedSearchAfterQuery);
+          return sampleDocSearchResultsNoSortId;
+        },
+        alertInstanceFactory: jest.fn(),
+        savedObjectsClient,
+      };
+      try {
+        const searchAfterResult = await singleSearchAfter(
+          searchAfterSortId,
+          sampleParams,
+          mockService,
+          mockLogger
+        );
+        expect(searchAfterResult).toBeEmpty();
+      } catch (exc) {
+        expect(exc.message).toEqual('Attempted to search after with empty sort id');
```

Comment:
When testing try catch blocks like this, you have a general problem which is that `expect` statement will only execute and fail the test if your function does a throw and then in the catch it sees a different value.

However, if your function of `singleSearchAfter` never does a throw, you never reach the `catch` block and never fail the test. Instead it continues and reports the test as still passing. You have what looks like an extra expect to catch that case well but there is one other way.

If you want to fail your test if you do _not_ throw then what you want to use is `rejects.toThrow`. Examples are in our code but this one would be:

```ts
      await expect(
        singleSearchAfter(searchAfterSortId, sampleParams, mockService, mockLogger)
      ).rejects.toThrow('Attempted to search after with empty sort id');
```

Then if that does not do a throw it will fail the test and you don't need to write extra expect statements for when it does not fail the test.